keps/sig-node/4800-cpumanager-split-uncorecache/README.md

```diff
@@ -0,0 +1,689 @@
+
+- Handle enablement of the policy option in pkg/kubelet/cm/cpumanager/policy_options.go and check the validity of the user flag and capability of the system.  If topology does not support split uncore caches, grouping by uncore cache static policy will be ignored.  
+
+- Modify the "Allocate" static policy to check for the option, align-cpus-by-uncorecache. For platforms where SMT is enabled, align-cpus-by-uncorecache will continue to follow default behaviour and try to allocate full cores when possible. Align-cpus-by-uncorecache can be enabled along with full-pcpus-only and enforce full core assignment with uncorecache alignment.
+
+- Align-cpus-by-uncorecache will be compatible with the default scheduling algorithm with future plans to be compatible with the options distribute-cpus-across-numa and distribute-cpus-across-cores.
+
+### User Stories (Optional)
+
+#### Story 1
+
+- As a HPC user, I want to extract the best performance possible and reduce latency, so that I can extract the most value when deploying applications in Kubernetes. I can use the static CPU Manager Policy option, but on split uncore cache processors, my application can experience latency when assigned CPUs across multiple uncore caches. In order to maximize performance, I want to minimize the distribution across uncore caches to minimize the latency. I want the scheduler to automatically handle this for me for larger clusters for simple deployment and efficiency.
+
+#### Story 2
+
+- As a Networking/Telco Engineer, my application is sensitive to uncore cache latency. I want my application to be statically assigned to CPUs that correspond to one single uncore cache so that I can get the maximum throughput from my application. Additionally, I have applications that are also static and of different CPU size requirements. The default Static CPU Manager Policy will result in my applications being assigned to two or more uncore caches. I have a large multi-node Kuberentes cluster. I want a scheduling policy to automatically assign CPUs for the best performance across the fewest amount of uncore cahces based on available CPUs without having to worry about what order/size I deploy my applications so I can ensure I get the best performance possible. 
+
+### Notes/Constraints/Caveats (Optional)
+
+- The name UnCoreCache is directly derived from cAdvisor that is being used as a package. 
+- Using the flow in `CPUManagerPolicy{Alpha,Beta}Options` from [xref](https://github.com/kubernetes/kubernetes/blob/master/pkg/features/kube_features.go#L110C1-L117C83) which is used to avoid a proliferation of a feature gates. 
+
+### Risks and Mitigations
+
+- Feature is enabled by a static policy option flag.  
+- It does not change behavior of non-static policy
+- It preserves the behavior of other static options.
+- Failure will occur if mismatch between options occurs and the Pod will not be scheduled.
```

Comment:
this is unclear though. A risk factor could be that since the feature is best-effort, low-latency workload can be admitted without ensuring optimal uncore cache alignment; but we will be adding metrics as mitigation factor so this can be detected.

Another obvious risk is bugs make the kubelet crash; a mitigation factor is twofold: we can disable the feature using the cpumanager policy option, and while non-GA we can disable with the feature gate.